pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -945,21 +951,29 @@ func (c *Cluster) assignMons(mons []*monConfig) error {
 }
 
 func (c *Cluster) monVolumeClaimTemplate(mon *monConfig) *v1.PersistentVolumeClaim {
-	if !c.spec.IsStretchCluster() {
-		return c.spec.Mon.VolumeClaimTemplate
-	}
 
-	// If a stretch cluster, a zone can override the template from the default.
-	for _, zone := range c.spec.Mon.StretchCluster.Zones {
-		if zone.Name == mon.Zone {
-			if zone.VolumeClaimTemplate != nil {
-				// Found an override for the volume claim template in the zone
-				return zone.VolumeClaimTemplate
+	if c.spec.IsStretchCluster() || len(c.spec.Mon.Zones) > 0 {
```

Comment:
This check for stretch cluster or zones is used in many places. How about creating a small helper method that will check this? For example:
```go
func zonesRequired(spec cephv1.ClusterSpec) bool {
    return spec.IsStretchCluster() || len(spec.Mon.Zones) > 0
}
```